acstools/acszpt.py

```diff
@@ -0,0 +1,379 @@
+        except ValueError:
+            result = '{} does not match YYYY-MM-DD format'.format(self.date)
+        else:
+            if dt_obj < self._acs_installation_date:
+                result = ('The observation date cannot occur '
+                          'before ACS was installed ({})'.
+                          format(self._acs_installation_date.strftime(fmt)))
+            elif dt_obj > self._extrapolation_date:
+                result = ('The observation date cannot occur after the '
+                          'maximum allowable date, {}. Extrapolations after '
+                          'this date invalid due to high uncertainties in the ' 
+                          'component and throughput files'.
+                          format(self._extrapolation_date.strftime(fmt)))
+        finally:
+            return result
+
+    def _submit_request(self):
+        """Submit a request to the ACS Zeropoint Calculator.
+
+        If an exception is raised during the request, an error message is
+        given. Otherwise, the response is saved in the corresponding
+        attribute.
+
+        """
+        try:
+            self._response = urlopen(self._url)
+        except URLError as e:
+            msg = ('{}\n{}\nThe query failed! Please check your inputs. '
+                   'If the error persists, submit a ticket to the '
+                   'ACS Help Desk at hsthelp.stsci.edu with the error message '
+                   'displayed above.'.format(str(e), self._msg_div))
+            LOG.error(msg)
+            self._failed = True
+        else:
+            self._failed = False
+
+    def _parse_and_format(self):
+        """ Parse and format the results returned by the ACS Zeropoint Calculator.
+
+        Using ``beautifulsoup4``, find all the ``<tb> </tb>`` tags present in
+        the response. Format the results into an astropy.table.QTable with
+        corresponding units and assign it to the zpt_table attribute.
+        """
+
+        soup = BeautifulSoup(self._response.read(), 'html.parser')
+
+        # Grab all elements in the table returned by the ZPT calc.
+        td = soup.find_all('td')
+
+        # Remove the units attached to PHOTFLAM and PHOTPLAM column names.
+        td = [val.text.split(' ')[0] for val in td]
+
+        # Turn the single list into a 2-D numpy array
+        data = np.reshape(td,
+                          (int(len(td) / self._block_size), self._block_size))
+        # Create the QTable, note that sometimes self._response will be empty
+        # even though the return was successful; hence the try/except to catch
+        # any potential index errors. Provide the user with a message and
+        # set the zpt_table to None.
+        try:
+            tab = QTable(data[1:, :], names=data[0],
+                       dtype=[str, float, float, float, float, float])
+        except IndexError as e:
+            msg = ('{}\n{}\nHmm, there was an issue parsing the request. '
```

Comment:
There is no need to insert sounds like "hmm" and "whoops".
```suggestion
            msg = ('{}\n{}\nThere was an issue parsing the request. '
```